### ISSUE TITLE
Useless variable is required for .each with object

### ISSUE THREAD

**Issue by tjchambers** [bbatsov/rubocop#1837]
```
app/lib/cd_reasoner_summary.rb:8:65: W: Useless assignment to variable - sum. Use just operator +.
    kind.where(assessment: @aa).each_with_object(0) { |cd, sum| sum += @cdr.reasoning_count(cd) }
                                                                ^^^
```

running on 

rubocop -V
0.29.1 (using Parser 2.2.0.3, running on ruby 2.2.0 x86_64-darwin14)

Suggests "Use just operator +". But I infer that the sum variable is required with this construction, so I am not sure how to avoid using it without rewriting the syntax to work around rubocop? Is this a bug, a limitation of rubocop or a syntax change I should make. Appreciate guidance.

**Comment by jonas054** [comment]
The above code line doesn't work. See the note about immutable objects here: http://apidock.com/rails/Enumerable/each_with_object

What you want is probably:
```ruby
kind.where(assessment: @aa).inject(0) { |sum, cd| sum + @cdr.reasoning_count(cd) }
```

Maybe we can update `Lint/UselessAssignment` to come with better advice in situations like this one.

**Comment by tjchambers** [comment]
Thanks for that education. I feel like a clearer explanation would have steered me off making this an issue and I would have been able to correct my error. Appreciate the insight. I concur about a more targeted cop.
